custom/icds_reports/static/js/directives/progress-report/progress-report.directive.js

```diff
@@ -130,11 +130,13 @@ function ProgressReportController($scope, $location, progressReportService,
 
     vm.getDisableIndex = function () {
         var i = -1;
-        window.angular.forEach(vm.selectedLocations, function (key, value) {
-            if (key !== null && key.location_id !== 'all' && !key.user_have_access) {
-                i = value;
-            }
-        });
+        if (!haveAccessToAllLocations) {
```

Comment:
can this be extracted into a shared function somehow? noticed what looked like almost the same logic pasted in a few places.